### PR TITLE
[spmd] set up package install for spmd

### PR DESCRIPTION
**Pull request by wanchaol**
This PR set up the package install for spmd package

to install it, in root folder of the repo, run:
`python spmd/setup.py install`

### CODE REVIEW

File: spmd/version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+0.1.0a0
```
Comment (wangkuiyi) [inline spmd/version.txt:1]: Starting with 0.0.1? :D

**Review by wangkuiyi — APPROVED** [review]
LGTM

**Review by fduwjj — APPROVED** [review]
lgtm

File: spmd/setup.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    sha: str = (
+        subprocess.check_output(["git", "rev-parse", "HEAD"], cwd=cwd)
+        .decode("ascii")
+        .strip()
+    )
+except Exception:
+    sha = "Unknown"
+
+if os.getenv("BUILD_VERSION"):
+    version = os.getenv("BUILD_VERSION", version)
```
Comment (yhcharles) [inline spmd/setup.py:32]: -> `version = os.getenv("BUILD_VERSION")`?
Since we already checked the env var is not None

**Review by yhcharles — APPROVED** [review]
LGTM

**Pull request merged**